tests/test_misc.py

```diff
@@ -397,7 +398,12 @@ def test_batched_cpfp_transaction(revault_network, bitcoind):
         assert txid in cpfp_entry["depends"]
 
     # Now get to be able to broadcast the second Spend.
-    bitcoind.generate_block(CSV, wait_for_mempool=unvaults)
+    bitcoind.generate_block(1, wait_for_mempool=unvaults)
+    # We mine them slowly to avoid race conditions
+    for i in range(0, CSV - 1):
+        bitcoind.generate_block(1)
+        time.sleep(1)
```

Comment:
Could you just get to `CSV - 1`, then mine `1` with `wait_for_mempool`?

Reply:
This `wait_for_mempool` doesn't work. You need to make sure to give revaultd to process height `CSV - 1` before mining the block `CSV`, otherwise you'll have the second spend broadcasted at height `CSV` AND cpfped in the same block.

Reply:
No, Spends are broadcasted *after* we check for CPFP. Therefore you can't have a Spend that it CPFPed in the same block as it is broadcast (although i think it'd conceptually make sense).